framework/tester.py

```diff
@@ -270,6 +274,16 @@ def start_all_servers(self):
 
     def start_tempesta(self):
         """ Start Tempesta and wait until the initialization process finish. """
+        # `unittest.TestLoader.discover` returns initialized objects, we can't
+        # raise `SkipTest` inside of `TempestaTest.__init__` because we are unable
+        # to interfere `unittest` code and catch that exception inside of it. Also,
+        # we cannot put it in `setUp` or any other method that is typically overloaded.
+        # Hence we put this logic here.
+        # After all, what Tempesta test would not require Tempesta to run?
+        if self._base:
+            self.skipTest("This is an abstract class")
```

Comment:
I think we can move this condition to `TempestaTest.setUp()`. In `setUp()` we call `__create_tempesta()` without which we can't call `start_tempesta()`. If someone override `setUp()` without calling base class, he anyway must re-implement some base logic, which includes **SkipTest** too.

Reply:
it's a bit unclear what should be done with the whole set of setUp() overrides, but - put this check in setUp() looks a little more correct for me - it looks like a test case init, so we must skip whole test case in right place